src/Featurizers/StringFeaturizer.h

```diff
@@ -88,9 +137,85 @@ class StringEstimator : public Components::InferenceOnlyEstimatorImpl<StringTran
 // |  StringTransformer
 // |
 // ----------------------------------------------------------------------
+template <typename T>
+StringTransformer<T>::StringTransformer(bool useEmptyStringsForNullValues /*=false*/) :
+    _useEmptyStringsForNullValues(
+        std::move(
+            [&useEmptyStringsForNullValues](void) -> bool & {
+                if(useEmptyStringsForNullValues && Traits<T>::IsNullableType == false)
+                    throw std::invalid_argument("empty strings cannot be used with types that are not nullable");
```

Comment:
Is it better to throw an error here or just default it back to the normal `ToString` approach?

Reply:
I prefer that the programmer be intentional about the behavior that they want when opting into less common scenarios.